Source/Core/VideoCommon/PerformanceMetrics.cpp

```diff
@@ -93,40 +124,41 @@ void PerformanceMetrics::DrawImGuiStats(const float backbuffer_scale) const
-                                                   1000.0 / 24.00,
-                                                   1000.0 / 20.00,
-                                                   1000.0 / 15.00,
-                                                   1000.0 / 10.00,
-                                                   1000.0 / 5.000,
-                                                   1000.0 / 2.000,
-                                                   1000.0,
-                                                   2000.0};
-
-      const DT vblank_time = m_vps_counter.GetDtAvg() + m_vps_counter.GetDtStd();
-      const DT frame_time = m_fps_counter.GetDtAvg() + m_fps_counter.GetDtStd();
+      static constexpr std::size_t num_ticks = 17;
+      static constexpr std::array<double, num_ticks> tick_marks = {0.0,
+                                                                   1000.0 / 360.0,
+                                                                   1000.0 / 240.0,
+                                                                   1000.0 / 180.0,
+                                                                   1000.0 / 120.0,
+                                                                   1000.0 / 90.00,
+                                                                   1000.0 / 59.94,
+                                                                   1000.0 / 40.00,
+                                                                   1000.0 / 29.97,
+                                                                   1000.0 / 24.00,
+                                                                   1000.0 / 20.00,
+                                                                   1000.0 / 15.00,
+                                                                   1000.0 / 10.00,
+                                                                   1000.0 / 5.000,
+                                                                   1000.0 / 2.000,
+                                                                   1000.0,
+                                                                   2000.0};
+
+      const DT vblank_time = m_vps_counter.GetDtAvg() + 2 * m_vps_counter.GetDtStd();
+      const DT frame_time = m_fps_counter.GetDtAvg() + 2 * m_fps_counter.GetDtStd();
       const double target_max_time = DT_ms(vblank_time + frame_time).count();
       const double a =
-          std::max(0.0, 1.0 - std::exp(-4000.0 * m_vps_counter.GetLastRawDt().count() /
-                                       DT_ms(m_vps_counter.GetSampleWindow()).count()));
+          std::max(0.0, 1.0 - std::exp(-4.0 * (DT_s(m_vps_counter.GetLastRawDt()) /
+                                               DT_s(m_vps_counter.GetSampleWindow()))));
 
-      static double max_time = 0.0;
-      if (std::isfinite(max_time))
-        max_time += a * (target_max_time - max_time);
+      static double s_graph_max_time = 0.0;
```

Comment:
Please don't hide a sneaky global state in the middle of the function here. This can just as well be a member variable of the `PerformanceMetrics` class.

Reply:
I pushed again with all of your changes